test/helpers.ts

```diff
@@ -0,0 +1,84 @@
+/// <reference path="../src/SBVRType.d.ts" />
+const chai = require('chai')
+chai.use(require('chai-as-promised'))
+chai.use(require('chai-datetime'))
+const types = require('../dist/index')
+import * as util from 'util'
+import * as _ from 'lodash'
+import { expect } from 'chai'
+
+export function runTest<I,O>(typeName: string, fn: (test: SBVRTypeTest<I,O>) => any) {
+	const type = types[typeName]
+	const test = (methodName: string, isAsync: boolean = true) => {
+		const method = (...args: any[]) => {
+			const inputs = args.slice(0, args.length - 1)
+			const callback = args[args.length - 1]
+			let _method = _.get(type, methodName)
+			if (isAsync) {
```

Comment:
Could you do this check outside of the function like it was before, rather than repeating it every time